### DISCUSSION TITLE
[code signing issue] Fastlane + GitHub Actions deployment to TestFlight

### DISCUSSION THREAD

**Discussion by kristupasxtm** [fastlane/fastlane#21581] [Q&A]
Hello fastlane community,
Our team has been battling the issue of code signing with fastlane for several days. We found many related issues online but none of the solutions worked. The error we are blocked by:

`/Users/runner/work/repo-name/repo-name/ios/Runner.xcodeproj: error: No profile for team '***' ***ing 'match Development com.abc.abcabc2343' found: Xcode couldn't find any provisioning profiles ***ing '***/match Development com.abc.abcabc2343'. Install the profile (by dragging and dropping it onto Xcode's dock item) or select a different one in the Signing & Capabilities tab of the target editor. (in target 'Runner' from project 'Runner')`

first of all, why is xcode trying to find `Development` Profile if Matchfile has type `appstore`? The distribution certificate and appstore profile is successfully installed by fastlane's `build_app` when github action runs. We also used `fastlane match appstore` to create prod certificate for github repo

here is a list of solutions we have tried:

- pass `allowProvisioningUpdates` to `build_app`
- add app specific profile to `provisioningProfiles` in `exportOptions`
- switch to manual signing in Xcode software, set profile in `signing & capabilities` and in `build settings`
- add development profile and cert to repo, it installs development profile but can't find it again 
- create distribution certificate with match from another Mac and other apple id
- combinations of above

the build is setup as in this example:
https://medium.com/@seshasai_30381/flutter-github-actions-a-ci-cd-pipeline-for-flutter-apps-with-github-actions-fastlane-and-7ca9a6b2a1cc

**Comment by DanushkaPerusinghe** [comment]
Same issue in React native with Fastlane in GitHub actions

`/Users/runner/work/*******/*****/ios/*****.xcodeproj: error: Provisioning profile "match AppStore **.******" doesn't include signing certificate "Apple Distribution: **** (*****)". (in target '******' from project '*******')`

**Comment by carlooshh** [comment]
Did you find a solution?

**Reply by kristupasxtm** [reply]
We used manual code signing instead of doing it with fastlane. First we had to install apple certificates and provisioning profiles on github worker. Here is a guide:
https://docs.github.com/en/actions/deployment/deploying-xcode-applications/installing-an-apple-certificate-on-macos-runners-for-xcode-development
Also, don't forget to set your xcode project to manual signing
Lastly, use fastlane to build the app

**Comment by digallardox** [comment]
apple DX is a joke

**Comment by vangelis-vargas** [comment]
We had a similar issue in our project and found that the solution mentioned [here](https://github.com/fastlane/fastlane/issues/14978#issuecomment-508113670) worked.  We had to add the following actions before the build step in the Fastfile:


```
    update_project_provisioning(
      xcodeproj: "Project.xcodeproj",
      use_automatic_signing: false,
      profile: "com.fastlane.example.#{options[:id]}.mobileprovision",
      build_configuration: "Release",
      code_signing_identity: "iPhone Distribution"
      path: "Project.xcodeproj",
      teamid: options[:team_id]
    )
```
   
Pay special care to the `code_signing_identity`.  In terminal run: `security find-identity -v -p codesigning` and use the distribution code sign identity listed.

**Comment by jackson26-source** [comment]
The "wrong profile type" symptom (match set to appstore but Xcode complains about a Development profile) almost always means at least one target in the project is still on Automatic signing, or the exportOptions plist isn't listing every target's bundle ID. A few things worth checking in order:

1. Automatic signing has to be turned off per-target, not just at the project level. If you have any extensions (widget, notification service, share extension, watch app, etc.), each one needs its own `update_project_provisioning` call or its own entry in the `targets:` array — a single call against the main target leaves the others on automatic signing, and Xcode falls back to whatever development profile it can find locally for those.
2. exportOptions.plist needs a `provisioningProfiles` dict entry for every single bundle ID being archived, not just the main app's. Missing one for an extension is the single most common cause of this exact "wrong profile type" error in CI specifically (works locally because Xcode auto-resolves it there, fails in CI because there's nothing to auto-resolve against).
3. If you changed match's profile type or regenerated certs recently, a stale profile can be cached in project.pbxproj from a prior manual-signing session. Worth diffing pbxproj's `PROVISIONING_PROFILE_SPECIFIER` values across all targets against what match actually installed, rather than assuming the whole project matches once one target is fixed.
For what it's worth, I built a free tool (Signing Doctor) specifically for the "which of these things is actually mismatched" problem — you point it at your provisioning profile, cert, and keystore and it tells you in plain English what's actually wrong (expired, wrong bundle ID, cert not in the profile, etc.) instead of Xcode's fairly opaque error text. github.com/jackson26-source/macless-signing-doctor if it's useful for narrowing this down.
